pkg/handlers/search_symbols.go

```diff
@@ -0,0 +1,68 @@
+package handlers
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"strconv"
+
+	"github.com/artmoskvin/hide/pkg/project"
+)
+
+const MinLimit = 1
+const MaxLimit = 100
+const DefaultLimit = 10
+
+type SearchSymbolsHandler struct {
+	pm project.Manager
+}
+
+func NewSearchSymbolsHandler(pm project.Manager) SearchSymbolsHandler {
+	return SearchSymbolsHandler{pm: pm}
+}
+
+func (h SearchSymbolsHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	projectID, err := getProjectID(r)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("invalid project ID: %s", err), http.StatusBadRequest)
+		return
+	}
+
+	query := r.URL.Query().Get("query")
+	if query == "" {
+		http.Error(w, "query not specified", http.StatusBadRequest)
+		return
+	}
+
+	limit := DefaultLimit
+
+	if limitParam := r.URL.Query().Get("limit"); limitParam != "" {
```

Comment:
I think we should use `Has()`, try to parse and through an error.  https://go.dev/play/p/Ech5tabjRP0

Reply:
Done